src/main/java/de/dhbw/karlsruhe/domain/wrappers/TimeWrapper.java

```diff
@@ -0,0 +1,45 @@
+package de.dhbw.karlsruhe.domain.wrappers;
+
+import java.util.concurrent.TimeUnit;
+
+public class TimeWrapper {
+
+    private String durationAsString;
+    private long durationInMillis;
+
+    public TimeWrapper(long durationInMillis) {
+        this.durationInMillis = durationInMillis;
+        this.durationAsString = this.millisToTime(durationInMillis);
+    }
+
+    public TimeWrapper(String durationRaw) {
+        String[] timesSeparated = durationRaw.split("(:|\\.)");
+        this.durationInMillis = this.timeToMillis(timesSeparated[0], timesSeparated[1], timesSeparated[2], timesSeparated[3]);
+    }
```

Comment:
Im ersten Konstruktor wird this.durationAsString gesetzt, hier nicht.

Eine Idee wäre auch die Variable this.durationAsString zu löschen und in getTimeAsString() die MEthode millistotime() aufzurufen. Sind dafür weniger Code Zeilen aber reduziert Komplexität.

Reply:
Die Komplexität wird dann durch das Entfernen der Variable reduziert, meinst du das so?
Ich persönlich finde es irgendwie relativ übersichtlich, wenn es noch eine Variable gibt.
Und so spart man sich einen Methodenaufruf, da man den String direkt hat und nicht erst konvertiert oder überseh ich etwas?

Reply:
Ich schaue es mir nachher auf jeden Fall nochmal an, dann komme ich bestimmt auf eine gute Lösung :)

Reply:
Ich finde so globale Variablen eher meistens unübersichtlicher.  Aber was ich gemeint habe ist, wenn mit dem zweiten Konstruktor das Objekt erstellt wird ist durationAsString nie gesetzt. du könntest also `this.durationAsString = this.millisToTime(durationInMillis); ` in den zweiten Konstruktor hinzufügen oder die Variable ganz los werden. Vielleicht ist durationraw im zweiten Konstruktor auch schon das gleiche Format, wie durationAsString, dann könntest du einfach den Wert zuweisen.

Reply:
Ach so war das gemeint.
Ja, ist das gleiche Format und wir jetzt so gesetzt.